guides/doc-Monitoring_Project/topics/proc_configuring-pcp-data-collection.adoc

```diff
@@ -0,0 +1,114 @@
+[id='configure-pcp-data-collection_{context}']
+= Configuring PCP Data Collection
+
+This procedure describes how to configure PCP to collect metrics about processes, {Project}, Apache HTTP Server, and PostgreSQL.
+
+.Procedure
+
+. To configure PCP to collect data about {Project} processes, create the `/var/lib/pcp/pmdas/proc/hotproc.conf` file and include the following content:
++
+----
+#pmdahotproc
+Version 1.0
+
+(
+  fname == "java" ||
+  fname == "redis-server" ||
+  (fname == "postgres" && psargs ~ /-D/) ||
+  psargs ~ /puma/ ||
+ifndef::foreman-el,foreman-deb[]
+  psargs ~ /pulpcore.app.wsgi:application/ ||
+  psargs ~ /pulpcore.content:server/ ||
+  psargs ~ /pulpcore-worker/ ||
+endif::[]
+  psargs ~ /sidekiq/ ||
+  psargs ~ /smart-proxy/
+)
+----
++
+By default, PCP collects basic system metrics. This step enables detailed metrics about the following {Project} processes:
++
+* Java
+* PostgreSQL
+* Redis
+* Dynflow
+* Puma
+ifndef::foreman-el,foreman-deb[]
+* Pulpcore
+endif::[]
+
+. Configure PCP to log the process metrics being collected.
++
+----
+# mkdir -p /var/lib/pcp/config/pmlogconf/foreman-hotproc
+# cat >/var/lib/pcp/config/pmlogconf/foreman-hotproc/summary << EOF
+#pmlogconf-setup 2.0
+ident   foreman hotproc metrics
+probe   hotproc.control.config != "" ? include : exclude
+        hotproc.psinfo.psargs
+        hotproc.psinfo.cnswap
+        hotproc.psinfo.nswap
+        hotproc.psinfo.rss
+        hotproc.psinfo.vsize
+        hotproc.psinfo.cstime
+        hotproc.psinfo.cutime
+        hotproc.psinfo.stime
+        hotproc.psinfo.utime
+        hotproc.io.write_bytes
+        hotproc.io.read_bytes
+        hotproc.schedstat.cpu_time
+        hotproc.fd.count
+EOF
+----
+
+. Install the process monitoring PMDA.
++
+----
+# cd /var/lib/pcp/pmdas/proc
+# ./Install
+----
+
+. Configure PCP to collect metrics from Apache HTTP Server.
++
+.. Enable the Apache HTTP Server extended status module.
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {foreman-installer} --enable-apache-mod-status
+----
+.. Enable the Apache HTTP Server PMDA.
++
+----
+# cd /var/lib/pcp/pmdas/apache
+# ./Install
+----
+
+. Configure PCP to collect metrics from PostgreSQL.
+
++
+----
+# cd /var/lib/pcp/pmdas/postgresql
+# ./Install
+----
+
+. Enable telemetry functionality in {Project}.
++
+[options="nowrap", subs="verbatim,quotes,attributes"]
+----
+# {foreman-installer} --foreman-telemetry-prometheus-enabled true
```

Comment:
This runs the installer multiple times. Would it be good to only run it once, saving time?

Reply:
FIXME: make the installer quicker :stuck_out_tongue_closed_eyes: